conventions/src/main/kotlin/io.opentelemetry.instrumentation.javaagent-shadowing.gradle.kts

```
@@ -17,11 +17,13 @@ tasks.withType<ShadowJar>().configureEach {
   // rewrite dependencies calling Logger.getLogger
   relocate("java.util.logging.Logger", "io.opentelemetry.javaagent.bootstrap.PatchLogger")
 
-  // prevents conflict with library instrumentation, since these classes live in the bootstrap class loader
-  relocate("io.opentelemetry.instrumentation", "io.opentelemetry.javaagent.shaded.instrumentation") {
-    // Exclude resource providers since they live in the agent class loader
-    exclude("io.opentelemetry.instrumentation.resources.*")
-    exclude("io.opentelemetry.instrumentation.spring.resources.*")
+  if (!project.hasProperty("disableShadowRelocate") || !(project.property("disableShadowRelocate").toString().toBoolean())) {
```

Nit: I think you can use `findProperty` to simplify this a bit:
```suggestion
  if (!(project.findProperty("disableShadowRelocate") as Boolean)) {
```

---

That is super weird -- we have numerous places where we use that kind of syntax, e.g. https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/aa2c2436c28919fa3d31cb426a08b7b99e983a6b/instrumentation/gwt-2.0/javaagent/build.gradle.kts#L35

Anyway, if it doesn't work it doesn't work, let's merge the version that compiles.

---

This property is different in that it isn't guaranteed to exist.  It will only be non-null if actually configured in your `gradle.properties` file. Maybe a default value could be defined somehow, but I'm not sure how to do that.

---

I've updated both PR's with a different simplification per @laurit's suggestion.